source/client/stream_decoder.cc

```diff
@@ -132,15 +132,21 @@ void StreamDecoder::onPoolReady(Envoy::Http::RequestEncoder& encoder,
     // we take the risk of erroneously reporting that we did send all the bytes, instead of always
     // reporting 0 bytes.
     stream_info_.addBytesReceived(request_body_size_);
-    // Revisit this when we have non-uniform request distributions and on-the-fly reconfiguration in
-    // place. The string size below MUST match the cap we put on RequestOptions::request_body_size
-    // in api/client/options.proto!
-    auto* fragment = new Envoy::Buffer::BufferFragmentImpl(
-        staticUploadContent().data(), request_body_size_,
-        [](const void*, size_t, const Envoy::Buffer::BufferFragmentImpl* frag) { delete frag; });
-    Envoy::Buffer::OwnedImpl body_buffer;
-    body_buffer.addBufferFragment(*fragment);
-    encoder.encodeData(body_buffer, true);
+    if (request_body_.empty()) {
```

Comment:
Currently we will only reach this point if `request_body_size_ > 0`, which is the case only if the user set the `Content-Length` header explicitly, or the `OptionsListRequestSource` set the `Content-Length` header from `request_body_size` in `RequestOptions` proto.

The outer `if` statement could be updated to

`if (request_body_size_ > 0 || !request_body_.empty())` 

so that the json body takes effect independent of the legacy `request_body_size_`/`Content-Length` mechanism.